src/components/App.js

```diff
@@ -76,38 +88,110 @@ class App extends Component<{}, State> {
     setNotes(copy)
   }
 
-  deleteNote = (): void => {
+  updateAndSave = (note: Note): void => {
+    const copy = this.state.notes.slice()
+    const index = _.findIndex(copy, { key: note.key })
+    if (index === -1) {
+      copy.splice(copy.length, 0, note)
+    } else {
+      copy.splice(index, 1, note)
+    }
+    setNotes(copy)
+
+    this.setState({
+      note: note,
+      notes: copy,
+    })
+  }
+
+  deleteNote = (note?: Note) => {
```

Comment:
One way to avoid repetition here without adding the boolean `save` argument that you had before is to use a function as a parameter. For example:

```
  updateActiveNote = (note: Note, onceFound: (Array<Note>)): void => {
     const copy = this.state.notes.slice()
     const index = _.findIndex(copy, { key: note.key })
     if (index === -1) {
       copy.splice(copy.length, 0, note)
     } else {
       copy.splice(index, 1, note)
     }

     onceFound(copy)

     this.setState({
      note: note,
      notes: copy,
     })
  }
```

Then the other method becomes:
```
updateAndSave = (note: Note): void => {
  updateActiveNote(note, (notesSubset) => {
    setNotes(notesSubset)
  })
}
```

You could even remove the extra anonymous function:
```
updateAndSave = (note: Note): void => {
  updateActiveNote(note, setNotes)
}
```